include/eve/detail/function/simd/x86/arithmetic_compounds.hpp

```diff
@@ -334,4 +334,34 @@ namespace eve::detail
       return self;
     }
   }
+
+  template<scalar_value T, value U, typename N>
+  EVE_FORCEINLINE decltype(auto) self_div(wide<T, N, x86_512_> &self, U const &other) noexcept
+      requires(scalar_value<U> || std::same_as<wide<T, N, x86_512_>, U>)
+  {
+    using type = wide<T, N, x86_512_>;
+
+    if constexpr( scalar_value<U> )
+    {
+      return self_div(self, type {other});
+    }
+    else if constexpr( std::same_as<type, U> )
+    {
+      if constexpr( std::same_as<T, double> )
+      {
+        self = _mm512_div_pd(self, other);
+      }
+      else if constexpr( std::same_as<T, float> )
+      {
+        self = _mm512_div_ps(self, other);
+      }
+      else
+      {
+        auto that = self;
+        self      = aggregate([](auto a, auto b) { return a /= b; }, that, other);
```

Comment:
What about `_mm512_div_epi64` and friends?
Or since they are truncating, they don't do the right thing?

What's with div by 0?

Reply:
Which intrinsics are you talking about ?
https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_div&expand=412,451,451,5867,2473,451&techs=AVX_512

Reply:
These ones
https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_div&expand=412,451,451,5867,2473,451